lib/matplotlib/projections/geo.py

```diff
@@ -269,24 +269,23 @@ def __init__(self, resolution):
             self._resolution = resolution
 
         def transform_non_affine(self, ll):
-            longitude = ll[:, 0:1]
-            latitude  = ll[:, 1:2]
+            longitude = ll[:, 0]
+            latitude = ll[:, 1]
 
             # Pre-compute some values
             half_long = longitude / 2.0
             cos_latitude = np.cos(latitude)
 
             alpha = np.arccos(cos_latitude * np.cos(half_long))
-            # Mask this array or we'll get divide-by-zero errors
-            alpha = ma.masked_where(alpha == 0.0, alpha)
-            # The numerators also need to be masked so that masked
-            # division will be invoked.
+            # Avoid divide-by-zero errors using same method as NumPy.
+            alpha[alpha == 0.0] = 1e-20
```

Comment:
`np.maximum(alpha, 1e-20, out=alpha)` should work better if you somehow have tiny alphas (between 0 and 1e-20) and also saves an extra allocation, I think.

Reply:
It's fast either way, but `np.maximum` takes more than twice as long, at least in my test with 10,000 pts.  I think the 1e-20 is a completely arbitrary small number, and its only purpose is to prevent division by exactly zero.  A tiny number still works:
```
In [21]: np.sin(1e-300) / 1e-300
1.0
```